use otp.c from google authenticator, tested to work

quoraquestions opened the pull request:
tested and working.

spectralabode committed:
fix otp

spectralabode committed:
Testcase otp

BenjaminSoelberg commented:
Great to see your pull request!
Cant wait to add it to my watch.

In +#define CONFIG_MOD_OTP_KEY "\xb6\x83\x70\x5f\x79\x9d\x59\x3f\x3f\x24\x03\x00\xb7\x38\xa0\x3c\x44\x6a\x94\x61”
this your real google auth key ?

And did you add the include path the in Commons.mk on purpose or was that an artefact of the build system?
Also a .bak fail was added to repo.

Regards
Benjamin


> On 6 Dec 2016, at 03.15, menehune <notifications@github.com> wrote:
> 
> tested and working.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19 <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19>
> Commit Summary
> 
> fix otp
> Testcase otp
> File Changes
> 
> M Common.mk <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19/files#diff-0> (2)
> M config.h <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19/files#diff-1> (6)
> M modinit.c <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19/files#diff-2> (1)
> M modules/otp.c <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19/files#diff-3> (543)
> A modules/otp.h <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19/files#diff-4> (40)
> A openchronos.dep.bak <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19/files#diff-5> (0)
> A testsalt.txt <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19/files#diff-6> (1)
> Patch Links:
> 
> https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19.patch <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19.patch>
> https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19.diff <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19.diff>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AA-4r1WytXoELBXsvouUBc4Hg8Cal573ks5rFMUsgaJpZM4LE8ih>.
>

quoraquestions commented:
Hi Benjamin,

That is salt from a throwaway account I made for testing, looks like make
clean keeps config.h.
I was seeing Lt of warnings about std include files hence I added full path
to the directory. the installation of gcc should actually have created
symlinks which it did not, hence the change to the file.

The back file is surely an artifact.

I have had success using fet debugger to successfully  attach to target via
mspdebug and break execution while watch shows 'boot', we can look at it
closely to see how to boot the watch without doing A reset after debugger
attach, I am not too familiar with how the boot ROM works at the moment. I
have a vbox VM to do all this on win10.


Btw the string in config.h is generated from bae32 decode, when you try it
out on your watch just paste the key that Google 2 factor provides into
make config nurses otp box by pressing shft-insert after you copy the key.


Regards,
M

On Dec 7, 2016 3:13 AM, "Benjamin Sølberg" <notifications@github.com> wrote:

Great to see your pull request!
Cant wait to add it to my watch.

In +#define CONFIG_MOD_OTP_KEY "\xb6\x83\x70\x5f\x79\x9d\x59\
x3f\x3f\x24\x03\x00\xb7\x38\xa0\x3c\x44\x6a\x94\x61”
this your real google auth key ?

And did you add the include path the in Commons.mk on purpose or was that
an artefact of the build system?
Also a .bak fail was added to repo.

Regards
Benjamin


> On 6 Dec 2016, at 03.15, menehune <notifications@github.com> wrote:
>
> tested and working.
>
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19 <
https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19>
> Commit Summary
>
> fix otp
> Testcase otp
> File Changes
>
> M Common.mk <https://github.com/BenjaminSoelberg/openchronos-
ng-elf/pull/19/files#diff-0> (2)
> M config.h <https://github.com/BenjaminSoelberg/openchronos-
ng-elf/pull/19/files#diff-1> (6)
> M modinit.c <https://github.com/BenjaminSoelberg/openchronos-
ng-elf/pull/19/files#diff-2> (1)
> M modules/otp.c <https://github.com/BenjaminSoelberg/openchronos-
ng-elf/pull/19/files#diff-3> (543)
> A modules/otp.h <https://github.com/BenjaminSoelberg/openchronos-
ng-elf/pull/19/files#diff-4> (40)
> A openchronos.dep.bak <https://github.com/BenjaminSoelberg/openchronos-
ng-elf/pull/19/files#diff-5> (0)
> A testsalt.txt <https://github.com/BenjaminSoelberg/openchronos-
ng-elf/pull/19/files#diff-6> (1)
> Patch Links:
>
> https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19.patch <
https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19.patch>
> https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19.diff <
https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19.diff>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/
BenjaminSoelberg/openchronos-ng-elf/pull/19>, or mute the thread <
https://github.com/notifications/unsubscribe-auth/AA-
4r1WytXoELBXsvouUBc4Hg8Cal573ks5rFMUsgaJpZM4LE8ih>.
>

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19#issuecomment-265421164>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AF-iXcJzywEnjcqajDzfEmYl-g3E5sRTks5rFpTzgaJpZM4LE8ih>
.

spectralabode committed:
Removed artifacts of build that are not cleaned during make clean

quoraquestions commented:
Hi Benjamin, 

Removed the .dep file and the testsalt also cleaned the otp key from the config.h in the latest commit to the forked off branch.

Regards,
M

spectralabode committed:
attempt to acquire tz from system

spectralabode committed:
remove config.h again

spectralabode committed:
add python based offline totp to test against watch codes

spectralabode committed:
Remove generic stuff such as UNRAVEL UNROLL from otp.c and otp.h, which does not apply to ezChronos

quoraquestions commented:
Thanks.

On Dec 27, 2016 10:51 AM, "Benjamin Sølberg" <notifications@github.com>
wrote:

> Merged #19
> <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BenjaminSoelberg/openchronos-ng-elf/pull/19#event-905929822>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF-iXQ8vF8xSfmb6TP9gAGudyqLEUQXFks5rMV4kgaJpZM4LE8ih>
> .
>
